Convert print/cout to use lsst.log tickets-dm/8547

parejkoj opened the pull request.

hsinfang commented on src/Associations.cc:
```diff
@@ -33,6 +34,8 @@ static double sqr(double x) {return x*x;}
 namespace lsst {
 namespace jointcal {
 
+static LOG_LOGGER _log = LOG_GET("jointcal.Associations");
```
Why not instantiate in an anonymous namespace, like how qserv does? 

Same questions in other files.

parejkoj replied:
Because I'd missed the part about the anonymous namespace in the earlier discussion. Please see the latest commit: is that more like what you want?

We definitely need DM-9418 to help with this sort of question.

hsinfang commented on src/ListMatch.cc:
```diff
@@ -406,33 +398,31 @@ SolList Solutions;
-	 std::cerr << "please store the involved images and contact the developpers" << std::endl;
+          LOGLS_ERROR(_log, "There is an internal inconsistency in ListMatchupRotShift.");
+          LOGLS_ERROR(_log, "maxContent  = " << maxContent);
+	      LOGLS_ERROR(_log, "matches->size() = " << a_list->size());
```
too many leading whitespaces

parejkoj replied:
Good catch: it was a tab lurking in there. I'm going to give it all a clang-format pass in the near future, which should get any others like this.

parejkoj commented on python/lsst/jointcal/dataIds.py:
```diff
@@ -59,22 +59,19 @@ def makeDataRefList(self, namespace):
             raise RuntimeError("Must call setDatasetType first")
         skymap = None
-        log = None
+        log = lsst.log.Log.getLogger("jointcal.dataIds")
```
Using namespace.log gives the log output as `root`, which doesn't seem as helpful when figuring out where the message came from. If that's the standard system though, I'll use it.

hsinfang replied:
That is weird; I thought the logger's name would be `_DefaultName`. 

I agree the `root` logger is not helpful, so do not object your way here.

parejkoj commented on python/lsst/jointcal/dataIds.py:
```diff
@@ -90,7 +87,6 @@ def makeDataRefList(self, namespace):
                     # Going with just the nearest tract.  Since we're throwing all tracts for the visit
                     # together, this shouldn't be a problem unless the tracts are much smaller than a CCD.
                     tract = skymap.findTract(wcs.pixelToSky(box.getCenter()))
-                    print("tract = ", tract)
```
Yes, these removals were intentional. Part of this logging cleanup was removing unneeded output (as determined by me: jointcal's original authors left a lot of debugging statements in place).

parejkoj commented on python/lsst/jointcal/jointcal.py:
```diff
@@ -399,19 +400,19 @@ def _fit_photometry(self, associations):
         fit.minimize("Model")
         chi2 = fit.computeChi2()
-        print(chi2)
+        self.log.info(str(chi2))
```
`chi2` actually gets converted into a useful string: `Chi2/ndof : 2310.63/728=3.17394`

parejkoj commented:
I was wondering about the double commits myself: My laptop and work desktop both have the same email in `.gitconfig, but one of them (until recently) had another blank field in the `.gitconfig`, which may have confused github (particularly when I rebased flattened a set of changes). We'll see if it goes away in the future, now that I've fixed that.
